app/elm/Update.elm

```diff
@@ -12,3 +15,42 @@ update msg model =
 
         ToggleCategory catId ->
             { model | categories = toggleCategory catId model.categories } ! []
+
+        SetDate date ->
+            { model | date = date } ! []
+
+        GetSearchResults ->
+            model ! [ decodeResults ]
+
+        SearchResults (Ok results) ->
+            { model | events = results } ! []
+
+        SearchResults (Err err) ->
+            let
+                log =
+                    Debug.log "Request Error" err
+            in
+                model ! []
+
+
+decodeResults : Cmd Msg
```

Comment:
I think all of this should be in a separate file, kind of like how we did the Data and Views (it might live in Data or somewhere else)